src/med_bench/mediation.py

```diff
@@ -106,159 +107,37 @@ def mediation_IPW(y, t, m, x, w, z, trim, logit, regularization=True, forest=Fal
-            for train_index, test_index in train_test_list:
-                if not forest:
-                    rf_x_clf = LogisticRegressionCV(Cs=cs, cv=CV_FOLDS)\
-                        .fit(x[train_index, :], t[train_index])
-                    rf_xm_clf = LogisticRegressionCV(Cs=cs, cv=CV_FOLDS)\
-                        .fit(np.hstack((x, m))[train_index, :], t[train_index])
-                else:
-                    rf_x_clf = RandomForestClassifier(n_estimators=100,
-                                                      min_samples_leaf=10)\
-                        .fit(x[train_index, :], t[train_index])
-                    rf_xm_clf = RandomForestClassifier(n_estimators=100,
-                                                       min_samples_leaf=10)\
-                        .fit(np.hstack((x, m))[train_index, :], t[train_index])
-                if calibration:
-                    p_x_clf = CalibratedClassifierCV(rf_x_clf,
-                                                     method=calib_method)\
-                        .fit(x[train_index, :], t[train_index])
-                    p_xm_clf = CalibratedClassifierCV(rf_xm_clf,
-                                                      method=calib_method)\
-                        .fit(np.hstack((x, m))[train_index, :], t[train_index])
-                else:
-                    p_x_clf = rf_x_clf
-                    p_xm_clf = rf_xm_clf
-                p_x[test_index] = p_x_clf.predict_proba(x[test_index, :])[:, 1]
-                p_xm[test_index] = p_xm_clf.predict_proba(
-                    np.hstack((x, m))[test_index, :])[:, 1]
-
-            # trimming. Following causal weight code, not sure I understand
-            # why we trim only on p_xm and not on p_x
-            ind = ((p_xm > trim) & (p_xm < (1 - trim)))
-            y, t, p_x, p_xm = y[ind], t[ind], p_x[ind], p_xm[ind]
-
-            # note on the names, ytmt' = Y(t, M(t')), the treatment needs to be
-            # binary but not the mediator
-            p_x = np.clip(p_x, clip, 1 - clip)
-            p_xm = np.clip(p_xm, clip, 1 - clip)
-
-            y1m1 = np.sum(y * t / p_x) / np.sum(t / p_x)
-            y1m0 = np.sum(y * t * (1 - p_xm) / (p_xm * (1 - p_x))) /\
-                np.sum(t * (1 - p_xm) / (p_xm * (1 - p_x)))
-            y0m0 = np.sum(y * (1 - t) / (1 - p_x)) /\
-                np.sum((1 - t) / (1 - p_x))
-            y0m1 = np.sum(y * (1 - t) * p_xm / ((1 - p_xm) * p_x)) /\
-                np.sum((1 - t) * p_xm / ((1 - p_xm) * p_x))
-
-            return(y1m1 - y0m0,
-                   y1m1 - y0m1,
-                   y1m0 - y0m0,
-                   y1m1 - y1m0,
-                   y0m1 - y0m0,
-                   np.sum(ind))
+    print(x)
+    classifier_x, classifier_xm = get_x_classifiers(regularization, forest, calibration, calib_method)
+    p_x, p_xm = estimate_px(t, m, x, crossfit, classifier_x, classifier_xm)
 
+    if z is not None:
+        raise NotImplementedError
```

Comment:
What is z ? Are we going to drop this ?  Should z simply be ignored ?

Reply:
I think if we drop it we need to change other parts of the code that invoke it, might be worth another mini PR, but agreed with you.

Reply:
This has been addressed in d87de6717f6941a27d8921eaddcaf1e917bdcd67. Unnecessary arguments are now removed.